iabtcf-gvl/src/main/java/com/iabtcf/gvl/v2/cache/GvlAsyncImplCache.java

```diff
@@ -0,0 +1,93 @@
+package com.iabtcf.gvl.v2.cache;
+
+/*-
+ * #%L
+ * iabtcf-gvl
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.iabtcf.gvl.v2.dao.GvlVendorDataMap;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * A reference implementation of GvlCache that builds cache in Asynchronous mode using the
+ * iab global vendor list url
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class GvlAsyncImplCache extends GvlAbstractCache {
+
+    public static final GvlAsyncImplCache INSTANCE = new GvlAsyncImplCache();
+
+    AsyncCache<Integer, Map<Integer, GvlVendorDataMap>> gvlCache = Caffeine.newBuilder()
+        .expireAfterWrite(396, TimeUnit.DAYS)
+        .maximumSize(200L)
+        .buildAsync();
+
```

Comment:
True, but a given version of GVL never changes, once its published. If there is a change, then the vendor list version HAS to change. So that's why this simplification will work.

Reply:
With a static HashMap, I don't want to deal with synchronization issues (if any). This caffeine cache is highly tuned for that.